dataplane/cosmos-db/client.go

```diff
@@ -0,0 +1,119 @@
+// Package cosmosdb provides clients for Microsoft Azure CosmosDb Services.
+package cosmosdb
```

Comment:
CosmosDB encompasses a variety of storage technologies.  Since this addition is strictly for MongoDB on Cosmos I think we should give the package a more meaningful name.

Reply:
I don't know. Each function let's you know you are getting a Mongo API client and it's within the cosmosdb package. Seems pretty solid. 

The other functions for constructing clients could be named similarly, but with their own protocol name.

Reply:
To be clear, I think they should all be packages under a cosmosbd top-level directory.  Something like this (names can use improvement but you get the idea).
```
cosmosdb/mongo
cosmosdb/docdb
cosmosdb/table
cosmosdb/graph
```
For storage we didn't segregate by technologies (blobs/files/queues etc) and it's a big mess, in terms of discoverability and maintenance.

Reply:
BTW, now that we have all APIs support, I think this should go in the services folder, not the dataplane folder.